frontend/src/metabase/api/utils/use-token-refresh.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect } from "react";
+
+import { Api, useGetSettingsQuery } from "metabase/api";
+import { useDispatch } from "metabase/lib/redux";
+
+const REFRESH_INTERVAL = 10 * 1000; // 10 seconds
+
+/**
+ * In some circumstances, a metabase instance may have a temporary token signalling that we
+ * should refresh session properties
```

Comment:
This is useful context but I think it could be a bit more elaborate and descriptive of the hook